### PR TITLE
feat: register package aliases

### PR DESCRIPTION
**Pull request by avegancafe**
Implements usage of `register_aliases` added in this PR: https://github.com/williamboman/mason.nvim/pull/1146

### CODE REVIEW

**Comment by avegancafe** [comment]
This initial commit is just the POC, let me know if this looks all good and I'll add all of the other LSPs to this table instead of how I did it in the other MR initially!

File: lua/mason-lspconfig/init.lua [hunk 1]
```diff
@@ -24,6 +24,10 @@ function M.setup(config)
         require "mason-lspconfig.ensure_installed"()
     end
 
+    require("mason-registry").register_aliases({
+        ['lua-language-server'] = { 'luals' }
+    })
```
Comment (williamboman) [inline lua/mason-lspconfig/init.lua:29]: Something like this ought to do it:

```suggestion
require("mason-registry").register_aliases(_.map(function(server_name)
    return { server_name }
end, require("mason-lspconfig.mappings.server").package_to_lspconfig))
```

**Review by williamboman — COMMENTED** [review]
lgtm

File: lua/mason-lspconfig/init.lua (hunk 1 above)
Comment (avegancafe) [reply lua/mason-lspconfig/init.lua]: Oh true I forgot about that mapping, done

**Review by williamboman — COMMENTED** [review]
It'd be nice with a test case for this, there should be a `setup_spec.lua` that we can add a test case to. I'm thinking something along the lines of

```lua
local stub = require("luassert.stub")
local mock = require("luassert.mock")

-- ...
it("should set up package aliases", function ()
  local registry = require("mason-registry")
  stub(registry, "set_package_aliases")

  require("mason-lspconfig.mappings.server").package_to_lspconfig = {
     ["rust-analyzer"] = "rust_analyzer",
     ["typescript-language-server"] = "tsserver"
  }

  mason_lspconfig.setup {}

  assert.spy(registry.set_package_aliases).was_called(1)
  assert.spy(registry.set_package_aliases).was_called_with {
    ["rust-analyzer"] = { "rust_analyzer" },
    ["typescript-language-server"] = { "tsserver" }
  }
end)
```

File: lua/mason-lspconfig/init.lua [hunk 2]
```diff
@@ -24,6 +24,10 @@ function M.setup(config)
         require "mason-lspconfig.ensure_installed"()
     end
 
+    require("mason-registry").register_package_aliases(_.map(function(server_name)
```
Comment (williamboman) [inline lua/mason-lspconfig/init.lua:27]: For backwards compatibility we need to check that this function exists before using it:

```lua
if registry.register_package_aliases then
  -- ...
end
```

**Comment by williamboman** [comment]
> Sounds good I'll add that test case. Out of curiosity, how do I actually run those tests? haha

```
$ FILE=<path_to_file> make test
```

or all tests

```
$ make test
```

If you're on Windows (without WSL) it's a bit harder (in that case just push things directly)

**Comment by williamboman** [comment]
Btw the test will fail on CI until `mason.nvim` is merged to main. This PR LGTM now!

**Comment by williamboman** [comment]
@williambotman /fixup

**Comment by williamboman** [comment]
@williambotman /fixup

**Comment by williamboman** [comment]
The tests are failing because we override the mappings in the new test (per my recommendation). Luassert doesn't provide a nice way to mock tables, I'll look into properly reverting the mock and push a fix!

**Comment by avegancafe** [comment]
Ohh I see haha sounds good. Let me know if you need anything!

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/tests/helpers/lua/test_helpers.lua b/tests/helpers/lua/test_helpers.lua
index 5e38639..15ccd5e 100644
--- a/tests/helpers/lua/test_helpers.lua
+++ b/tests/helpers/lua/test_helpers.lua
@@ -16,6 +16,25 @@ function async_test(suspend_fn)
     end
 end
 
+local TableMock = {}
+TableMock.__index = TableMock
+
+function TableMock.new(tbl, key)
+    return setmetatable({
+        tbl = tbl,
+        key = key,
+    }, TableMock)
+end
+
+function TableMock:apply(value)
+    self.original_value = self.tbl[self.key]
+    self.tbl[self.key] = value
+end
+
+function TableMock:revert()
+    self.tbl[self.key] = self.original_value
+end
+
 -- selene: allow(incorrect_standard_library_use)
 mockx = {
     just_runs = function() end,
@@ -29,6 +48,12 @@ mockx = {
             error(exception, 2)
         end
     end,
+    table = function(tbl, key, new_value)
+        local mock = TableMock.new(tbl, key)
+        mock:apply(new_value)
+        assert:add_spy(mock)
+        return mock
+    end,
 }
 
 ---@param package_name string
diff --git a/tests/mason-lspconfig/setup_spec.lua b/tests/mason-lspconfig/setup_spec.lua
index 1cbd9e6..6c8f293 100644
--- a/tests/mason-lspconfig/setup_spec.lua
+++ b/tests/mason-lspconfig/setup_spec.lua
@@ -229,10 +229,10 @@ describe("mason-lspconfig setup", function()
     it("should set up package aliases", function()
         stub(registry, "register_package_aliases")
 
-        require("mason-lspconfig.mappings.server").package_to_lspconfig = {
+        local mapping_mock = mockx.table(require "mason-lspconfig.mappings.server", "package_to_lspconfig", {
             ["rust-analyzer"] = "rust_analyzer",
             ["typescript-language-server"] = "tsserver",
-        }
+        })
 
         mason_lspconfig.setup {}
 
@@ -241,6 +241,7 @@ describe("mason-lspconfig setup", function()
             ["rust-analyzer"] = { "rust_analyzer" },
             ["typescript-language-server"] = { "tsserver" },
         }
+        mapping_mock:revert()
     end)
 end)
 
```

File: lua/mason-lspconfig/init.lua [hunk 3]
```diff
@@ -24,6 +24,13 @@ function M.setup(config)
         require "mason-lspconfig.ensure_installed"()
     end
 
+    local registry = require "mason-registry"
+    if registry.register_package_aliases then
+        require("mason-registry").register_package_aliases(_.map(function(server_name)
```
Comment (williamboman) [inline lua/mason-lspconfig/init.lua:29]:
```suggestion
        registry.register_package_aliases(_.map(function(server_name)
```

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/vim.yml b/vim.yml
index 72b86e7..2fb8959 100644
--- a/vim.yml
+++ b/vim.yml
@@ -31,6 +31,11 @@ globals:
     mockx.returns:
         args:
             - type: any
+    mockx.table:
+        args:
+            - type: table
+            - type: string
+            - type: any
 
     describe:
         args:
```

**Review by williamboman — APPROVED** [review]
Thanks again and sorry for all the spam across the 2 PRs, should be all good now 😅!

**Comment by avegancafe** [comment]
Haha all good nice! Thanks for the review 😄

**Pull request merged**